soh/soh/Enhancements/randomizer/ShuffleRocks.cpp

```diff
@@ -0,0 +1,630 @@
+        return;
+
+    ObjBombiwa* rockActor = static_cast<ObjBombiwa*>(actorRef);
+
+    rockActor->rockIdentity = OTRGlobals::Instance->gRandomizer->IdentifyRock(
+        gPlayState->sceneNum, (s16)actor->world.pos.x, (s16)actor->world.pos.z);
+    if (rockActor->rockIdentity.randomizerCheck == RC_MAX) {
+        LUSLOG_INFO("ROCK bombiwa\t:\t%d, %d", rockActor->rockIdentity.randomizerCheck, (s16)actor->world.pos.x,
+                    (s16)actor->world.pos.z);
+    } else {
+        LUSLOG_INFO("ROCK bombiwa%d\t:\t%d, %d", rockActor->rockIdentity.randomizerCheck, (s16)actor->world.pos.x,
+                    (s16)actor->world.pos.z);
+    }
+    if (Rock_RandomizerHoldsItem(rockActor->rockIdentity, gPlayState, true)) {
+        rockActor->actor.draw = ObjBombiwa_RandomizerDraw;
+    }
+}
+
+void ObjHamishi_RandomizerInit(void* actorRef) {
+    Actor* actor = static_cast<Actor*>(actorRef);
+
+    if (actor->id != ACTOR_OBJ_HAMISHI)
+        return;
+
+    ObjHamishi* rockActor = static_cast<ObjHamishi*>(actorRef);
+
+    rockActor->rockIdentity = OTRGlobals::Instance->gRandomizer->IdentifyRock(
+        gPlayState->sceneNum, (s16)actor->world.pos.x, (s16)actor->world.pos.z);
+    if (rockActor->rockIdentity.randomizerCheck == RC_MAX) {
+        LUSLOG_WARN("ROCK hamishi\t:\t%d, %d", rockActor->rockIdentity.randomizerCheck, (s16)actor->world.pos.x,
+                    (s16)actor->world.pos.z);
+    } else {
+        LUSLOG_INFO("ROCK hamishi%d\t:\t%d, %d", rockActor->rockIdentity.randomizerCheck, (s16)actor->world.pos.x,
+                    (s16)actor->world.pos.z);
+    }
+    if (Rock_RandomizerHoldsItem(rockActor->rockIdentity, gPlayState, true)) {
+        rockActor->actor.draw = ObjHamishi_RandomizerDraw;
+    }
+}
+
+void ObjBombiwa_RandomizerKill(void* actorRef) {
+    Actor* actor = static_cast<Actor*>(actorRef);
+
+    if (actor->id != ACTOR_OBJ_BOMBIWA)
+        return;
+
+    ObjBombiwa* rockActor = static_cast<ObjBombiwa*>(actorRef);
+
+    if (Rock_RandomizerHoldsItem(rockActor->rockIdentity, gPlayState, true)) {
+        Rock_RandomizerSpawnCollectible(&rockActor->actor, rockActor->rockIdentity, gPlayState);
+        rockActor->rockIdentity.randomizerCheck = RC_MAX;
+        rockActor->rockIdentity.randomizerInf = RAND_INF_MAX;
+    }
+}
+
+void ObjHamishi_RandomizerKill(void* actorRef) {
+    Actor* actor = static_cast<Actor*>(actorRef);
+
+    if (actor->id != ACTOR_OBJ_HAMISHI)
+        return;
+
+    ObjHamishi* rockActor = static_cast<ObjHamishi*>(actorRef);
+
+    if (Rock_RandomizerHoldsItem(rockActor->rockIdentity, gPlayState, true)) {
+        Rock_RandomizerSpawnCollectible(&rockActor->actor, rockActor->rockIdentity, gPlayState);
+        rockActor->rockIdentity.randomizerCheck = RC_MAX;
+        rockActor->rockIdentity.randomizerInf = RAND_INF_MAX;
+    }
+}
+
+void RegisterShuffleRock() {
+    bool shouldRegister = IS_RANDO && (RAND_GET_OPTION(RSK_SHUFFLE_ROCKS) || RAND_GET_OPTION(RSK_SHUFFLE_ROCKS));
```

Comment:
Duplicated condition here means you never get the checks from boulders if you have shuffle boulders on but shuffle rocks off.